src/geoarray.jl

```diff
@@ -1,22 +1,95 @@
+"""
+    GeoArray{T <: Union{Real,Union{Missing,Real}}} <: AbstractArray{T,3}
+
+A GeoArray is an AbstractArray, an AffineMap for calculating coordinates based on the axes and a CRS definition to interpret these coordinates into in the real world.
+It's three dimensional and can be seen as a stack (3D) of 2D geospatial rasters (bands), the dimensions are :x, :y, and :bands.
+The AffineMap and CRS (coordinates) only operate on the :x and :y dimensions.
+"""
 mutable struct GeoArray{T <: Union{Real,Union{Missing,Real}}} <: AbstractArray{T,3}
     A::AbstractArray{T,3}
     f::AffineMap
     crs::WellKnownText{GeoFormatTypes.CRS,<:String}
```

Comment:
Hmm none of the three fields are concrete types, right? Is there a reason you went with that? Since normally it's best to avoid such that the compiler can specialize: https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type

Reply:
A is either an ArchGDAL IDataSet or an Array{T}.
AffineMap just isn't concrete, it's subs (linear and translation can be any kind of array, often StaticArrays, but not always).
WellKnownText also isn't concrete.

I'm happy to take suggestions how to resolve this.